### ISSUE TITLE
App Crash in ios 13 version.

### ISSUE THREAD

**Issue by priyasamyal** [terikon/cordova-plugin-photo-library#179]
Following are the logs generated in iOS version 13. Works fine in previous pervious. Please update ASAP.
Unexpectedly found nil while unwrapping an Optional value: file /Volumes/Data/Projects/Destinate/destinate/platforms/ios/destinate/Plugins/cordova-plugin-photo-library/PhotoLibraryService.swift, line 317
2019-09-30 12:36:26.209539+0530 destinate[503:147761] Fatal error: Unexpectedly found nil while unwrapping an Optional value: file /Volumes/Data/Projects/Destinate/destinate/platforms/ios/destinate/Plugins/cordova-plugin-photo-library/PhotoLibraryService.swift, line 317

**Comment by xmarkclx** [comment]
I'm using my fork with iOS 13.
There are instructions to use it as a replacement for this one if you are interested:

https://github.com/xmarkclx/cordova-plugin-photo-library-sism

**Comment by iamrafo** [comment]
@xmarkclx your fork not working with ios13.1

**Comment by xmarkclx** [comment]
> @xmarkclx your fork not working with ios13.1

That's really odd, I was just testing it yesterday, and am in fact using it right now on an iOS 13 iPhone and iPadOS 13 iPad Pro.

Are you sure you removed the original photo library?

What error are you getting?

**Comment by iamrafo** [comment]
> > @xmarkclx your fork not working with ios13.1
> 
> That's really odd, I was just testing it yesterday, and am in fact using it right now on an iOS 13 iPhone and iPadOS 13 iPad Pro.
> 
> Are you sure you removed the original photo library?
> 
> What error are you getting?

yes i'm sure i removed original library and use yours, and there is no errors but plugin does not getting images anyway

**Comment by xmarkclx** [comment]
Oh, i only use the save image and get permissions function, we used a
custom designed gallery for everything else.

**Comment by iamrafo** [comment]
i solve this issue by replacing ` 

> 

let file_url:URL = info!["PHImageFileURLKey"] as! URL
completion(file_url.relativePath, libraryItem)

> 

to

> 

let file_url:NSString = (info![“PHImageFileUTIKey”] as? NSString)!
completion(file_url as String, libraryItem)

> 

in the 314 line of PhotoLibraryService.swift file

PHImageFileURLKey deprecated in ios 13, i removed it by PHImageFileUTIKey

**Comment by iamrafo** [comment]
i solve this issue by replacing `
`let file_url:URL = info!["PHImageFileURLKey"] as! URL
completion(file_url.relativePath, libraryItem)`
to
`let file_url:NSString = (info![“PHImageFileUTIKey”] as? NSString)!
completion(file_url as String, libraryItem)`
in the 314 line of PhotoLibraryService.swift file

PHImageFileURLKey deprecated in ios 13,  instead use PHImageFileUTIKey

**Comment by iamrafo** [comment]
i solve this issue by replacing

_**let file_url:URL = info!["PHImageFileURLKey"] as! URL
completion(file_url.relativePath, libraryItem)**_
to

**_let file_url:NSString = (info![“PHImageFileUTIKey”] as? NSString)!
completion(file_url as String, libraryItem)_**

in the 314 line of PhotoLibraryService.swift file

PHImageFileURLKey deprecated in ios 13, instead use PHImageFileUTIKey

**Comment by xmarkclx** [comment]
@iamrafo Ok, added that fix on 2.2.95

**Comment by abcoskn** [comment]
> i solve this issue by replacing
> 
> _**let file_url:URL = info!["PHImageFileURLKey"] as! URL completion(file_url.relativePath, libraryItem)**_ to
> 
> **_let file_url:NSString = (info![“PHImageFileUTIKey”] as? NSString)! completion(file_url as String, libraryItem)_**
> 
> in the 314 line of PhotoLibraryService.swift file
> 
> PHImageFileURLKey deprecated in ios 13, instead use PHImageFileUTIKey

You saved my day, thank you!
